Simple-Projects/Simple_Projects/Basic/List/List.cpp

```diff
@@ -58,11 +52,9 @@ int List::find(int value) const {
 }
 
 void List::push_back(int value) {
-    int newCurrent = current + 1;
-
-    if ( newCurrent > capacity ) {
-       int newCapacity = capacity * multiplier;
-       int* newArray = (int*)realloc(array, newCapacity*sizeof(int));
+    if ( current == capacity ) {
+       const int newCapacity = capacity * multiplier;
+       int* const newArray = (int*)realloc(array, newCapacity*sizeof(int));
```

Comment:
realloc? Nice :)
But note that realloc will only work correctly (copying the contents) when you work with POD (Plain Old Data) types.
But objects with custom copy/move semantics will need to have the respective copy/move constructors called, which realloc does not do.